src/rm/rm.rs

```diff
@@ -129,7 +129,7 @@ fn remove(files: Vec<String>, force: bool, interactive: InteractiveMode, one_fs:
         let filename = &filename[..];
         let file = Path::new(filename);
         if file.exists() {
-            if file.is_dir() {
+            if file.symlink_metadata().unwrap().is_dir() {
```

Comment:
I think both exist() and symlink_metadata() fail when permission of the parent directory doesn't have 'X' permission. Anyway, I agree unrwap() in my first patch didn't look good. Thanks for your review and merge!